aiida_quantumespresso/parsers/ph.py

```diff
@@ -16,44 +18,85 @@ class PhParser(Parser):
     def parse(self, **kwargs):
         """Parse the retrieved files from a `PhCalculation`."""
         try:
-            output_folder = self.retrieved
+            self.retrieved
         except exceptions.NotExistent:
-            return self.exit_codes.ERROR_NO_RETRIEVED_FOLDER
-
-        list_of_files = output_folder.list_object_names()
+            return self.exit(self.exit_codes.ERROR_NO_RETRIEVED_FOLDER)
 
         # The stdout is required for parsing
         filename_stdout = self.node.get_attribute('output_filename')
         filename_tensor = PhCalculation._OUTPUT_XML_TENSOR_FILE_NAME
         filepath_dynmat = PhCalculation._FOLDER_DYNAMICAL_MATRIX
 
-        if filename_stdout not in list_of_files:
-            self.logger.error("The standard output file '{}' was not found but is required".format(filename_stdout))
-            return self.exit_codes.ERROR_READING_OUTPUT_FILE
+        if filename_stdout not in self.retrieved.list_object_names():
+            return self.exit_codes.ERROR_OUTPUT_STDOUT_MISSING
 
-        xml_tensor_file = None
-        if filename_tensor in list_of_files:
-            xml_tensor_file = output_folder._repository._get_base_folder().get_abs_path(filename_tensor)
+        try:
+            stdout = self.retrieved.get_object_content(filename_stdout)
+        except IOError:
```

Comment:
Should we also catch `OSError`?
https://stackoverflow.com/questions/29347790/difference-between-ioerror-and-oserror

Reply:
Agreed